app.rb

```diff
@@ -0,0 +1,166 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    puts 'Welcome to the School Library App!'
+    loop do
+      display_menu
+      choice = gets.chomp
+
+      if MENU_OPTIONS.include?(choice)
+        send(MENU_OPTIONS[choice])
+      else
+        puts 'Invalid choice. Please try again.'
+      end
+    end
+  end
+
+  def display_menu
+    puts 'Please choose an option:'
+    MENU_OPTIONS.each do |key, value|
+      puts "#{key}. #{value.to_s.tr('_', ' ')}"
+    end
+  end
+
```

Comment:
As per project requirements, kindly make sure that the `main.rb` file has the code that is responsible for presenting the user with the list of options to chose from and invokes the actions.

![image](https://github.com/Eccb7/school-library/assets/38088818/34830500-5faa-4a6c-ad67-b718b3f581cb)

- In your case the "MENU_OPTIONS", `start` and `display_menu` menu methods in the `app.rb` file should be moved to the `main.rb` since they are responsible for performing user actions.

You should have something similar to this:

```ruby
# main.rb
def main
  app = App.new()

  response = nil

  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"

    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

# 'main' method should be invoked at the end of the main.rb file
main()
```